ops/op2/object_wrap.rs

```diff
@@ -0,0 +1,131 @@
+// Copyright 2018-2024 the Deno authors. All rights reserved. MIT license.
+
+use proc_macro2::TokenStream;
+use quote::quote;
+use quote::ToTokens;
+use syn::ImplItem;
+use syn::ItemFn;
+use syn::ItemImpl;
+
+use crate::op2::generate_op2;
+use crate::op2::MacroConfig;
+use crate::op2::Op2Error;
+
+// Object wrap for Cppgc-backed objects
+//
+// This module generates the glue code declarations
+// for `impl` blocks to create JS objects in Rust
+// using the op2 infra.
+//
+// ```rust
+// #[op]
+// impl MyObject {
+//    #[constructor] // <-- first attribute defines binding type
+//    #[cppgc]       // <-- attributes for op2
+//    fn new() -> MyObject {
+//      MyObject::new()
+//    }
+//
+//    #[static_method]
+//    #[cppgc]
+//    fn static_method() -> MyObject {
+//      // ...
+//    }
+//
+//    #[method]
+//    #[smi]
+//    fn x(&self) -> i32 {
+//      // ...
+//    }
+// }
+//
+// The generated OpMethodDecl that can be passed to
+// `deno_core::extension!` macro to register the object
+//
+// ```rust
+// deno_core::extension!(
+//   ...,
+//   objects = [MyObject],
+// )
+// ```
+//
+// ```js
+// import { MyObject } from "ext:core/ops";
```

Comment:
Feels somewhat odd that we import it from `ext:core/ops`, fine for the first pass though 👍